src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2294,21 +2292,21 @@ protected virtual bool MarkMethods (TypeDefinition type, in DependencyInfo reaso
 		void MarkMethodCollection (IList<MethodDefinition> methods, in DependencyInfo reason, IMemberDefinition sourceLocationMember)
 		{
 			foreach (MethodDefinition method in methods)
-				MarkMethod (method, reason, sourceLocationMember);
+				MarkMethod (method, reason, new MessageOrigin (sourceLocationMember));
 		}
 
 		protected internal void MarkIndirectlyCalledMethod (MethodDefinition method, in DependencyInfo reason, IMemberDefinition sourceLocationMember)
 		{
-			MarkMethod (method, reason, sourceLocationMember);
+			MarkMethod (method, reason, new MessageOrigin (sourceLocationMember));
 			Annotations.MarkIndirectlyCalledMethod (method);
 		}
 
-		protected virtual MethodDefinition MarkMethod (MethodReference reference, DependencyInfo reason, IMemberDefinition sourceLocationMember)
+		protected virtual MethodDefinition MarkMethod (MethodReference reference, DependencyInfo reason, MessageOrigin sourceLocationMember)
```

Comment:
I don't expect this to have a big impact, but can you please roughly measure that this doesn't slow linker down?

I would also rename the parameter to something like `origin` or `markOrigin` or `sourceLocation` or... I don't know... it's definitely not `sourceLocationMember`.

Reply:
Overall, the linker takes relatively the same time for trimming a template hello world app (the linker that I build locally is actually 1 sec faster than the one shipped with the SDK, but this is unrelated to this change - 4.5s vs 5.5s). I can see `ProcessMethod` taking 2% more time (12.5% of total linking time, compared to the current 10.5%.) Currently, MarkMethod gets inlined - with this change this is no longer the case.